### PR TITLE
Remove existing OpenTelemetry Collector code

### PR DESCRIPTION
**Pull request by jpkrohling**
The current OpenTelemetry code isn't representative anymore of where we want to go for Jaeger v2. In order to reduce confusion to users, we discussed during the last weekly meeting about removing this code. Exporters that were created for this are still available in earlier tags and can be recovered from there when needed.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by pavolloffay** [comment]
Are these components going to be migrated to  https://github.com/jaegertracing/jaeger-otelcol? 

The main reason for having the code here was for the ease of development. Many times changes in the core Jaeger components were required.

**Comment by joe-elliott** [comment]
There were a number of places that these "AddOtelFlags" methods were added:

https://github.com/jaegertracing/jaeger/blob/master/cmd/ingester/app/flags.go#L92

Should we clean those up as well?

**Comment by jpkrohling** [comment]
From what I understand, we'll first adapt this repository here to have multiple go modules, and then move the contents of jaeger-otelcol to this repo here. The goal is to avoid the circular dependency between otelcol and Jaeger, while still being able to have the binaries being built as part of this single repository.

@joe-elliott is working on the reorganization, and @kevinearls will work on moving things here. Some of the components removed by this PR will get a new life after this reorganization.

**Comment by jpkrohling** [comment]
For context, the decision to remove this here right now also has to do with #2781, as we can't bump Thrift to 0.14 in the otelcol modules as it depends on a Zipkin Thrift repository that is still on 0.13. Unfortunately, Thrift 0.14 is incompatible with 0.13, as it adds `context.Context` to some methods...

**Comment by jpkrohling** [comment]
> Should we clean those up as well?

Done!

**Comment by pavolloffay** [comment]
> For context, the decision to remove this here right now also has to do with #2781, as we can't bump Thrift to 0.14 in the otelcol modules as it depends on a Zipkin Thrift repository that is still on 0.13. Unfortunately, Thrift 0.14 is incompatible with 0.13, as it adds context.Context to some methods...

The same problem might pop up after in OTEL collector as well. Better would be to bump zipkin to thrift 0.14 as well.

**Comment by jpkrohling** [comment]
> The same problem might pop up after in OTEL collector as well.

Indeed, it will affect that as well.

**Review by joe-elliott — APPROVED** [review]
Looks good to me

**Review by joe-elliott — CHANGES_REQUESTED** [review]
Apologies, found a few more non-blocking otel references after approving:

https://github.com/jaegertracing/jaeger/blob/master/.gitignore#L37
https://github.com/jaegertracing/jaeger/blob/master/Makefile#L3
https://github.com/jaegertracing/jaeger/blob/master/Makefile#L177
https://github.com/jaegertracing/jaeger/blob/master/scripts/upload-all-docker-images.sh#L33
https://github.com/jaegertracing/jaeger/blob/master/scripts/upload-to-registry.sh#L84

File: scripts/build-all-in-one-image.sh [hunk 1]
```diff
@@ -67,9 +67,3 @@ docker build -f cmd/all-in-one/Dockerfile \
         --build-arg TARGETARCH=$GOARCH
 run_integration_test $repo
 upload_to_docker $repo
-
-make build-otel-all-in-one GOOS=linux GOARCH=$GOARCH
-repo=jaegertracing/opentelemetry-all-in-one
-docker build -f cmd/opentelemetry/cmd/all-in-one/Dockerfile -t $repo:latest cmd/opentelemetry/cmd/all-in-one --build-arg TARGETARCH=$GOARCH
-run_integration_test $repo
-upload_to_docker $repo
```
Comment (yurishkuro) [inline scripts/build-all-in-one-image.sh:75]: there's more stuff in scripts:
```
$ grx opentelemetry scripts
scripts/upload-all-docker-images.sh:33:	opentelemetry-collector
scripts/upload-all-docker-images.sh:34:	opentelemetry-agent
scripts/upload-all-docker-images.sh:35:	opentelemetry-ingester
scripts/build-all-in-one-image.sh:72:repo=jaegertracing/opentelemetry-all-in-one
scripts/build-all-in-one-image.sh:73:docker build -f cmd/opentelemetry/cmd/all-in-one/Dockerfile -t $repo:latest cmd/opentelemetry/cmd/all-in-one --build-arg TARGETARCH=$GOARCH
scripts/upload-to-registry.sh:84:      "jaegertracing/jaeger-opentelemetry-collector" | "jaegertracing/jaeger-opentelemetry-agent" | "jaegertracing/jaeger-opentelemetry-ingester" | "jaegertracing/opentelemetry-all-in-one" )
```

Comment (jpkrohling) [reply scripts/build-all-in-one-image.sh:75]: Done

File: cmd/agent/app/reporter/grpc/flags.go [hunk 2]
```diff
@@ -41,11 +41,6 @@ var tlsFlagsConfig = tlscfg.ClientFlagsConfig{
 func AddFlags(flags *flag.FlagSet) {
 	flags.Uint(retry, defaultMaxRetry, "Sets the maximum number of retries for a call")
 	flags.Int(discoveryMinPeers, 3, "Max number of collectors to which the agent will try to connect at any given time")
-	AddOTELFlags(flags)
-}
-
-// AddOTELFlags adds flags that are exposed by OTEL collector
-func AddOTELFlags(flags *flag.FlagSet) {
 	flags.String(collectorHostPort, "", "Comma-separated string representing host:port of a static list of collectors to connect to directly")
```
Comment (albertteoh) [inline cmd/agent/app/reporter/grpc/flags.go:44]: I wonder why these were considered specific to OTEL in the first place

**Review by albertteoh — APPROVED** [review]
lgtm

File: cmd/agent/app/reporter/grpc/flags.go (hunk 2 above)
Comment (pavolloffay) [reply cmd/agent/app/reporter/grpc/flags.go:44]: Not all flags were added to OTEL distros in other words some flags made sense only for legacy Jaeger.

**Review by joe-elliott — APPROVED** [review]

**Comment by jpkrohling** [comment]
I'm merging with two approvals, any left overs can be removed in a follow-up PR.

**Pull request merged**